phoenix-core/src/main/java/org/apache/phoenix/compile/FromCompiler.java

```diff
@@ -383,6 +448,85 @@ protected TableRef createTableRef(NamedTableNode tableNode, boolean updateCacheI
             return tableRef;
         }
 
+        @Override
+        public List<PFunction> getFunctions() {
+            return functions;
+        }
+
+        protected List<PFunction> createFunctionRef(List<String> functionNames, boolean updateCacheImmediately) throws SQLException {
```

Comment:
Minor nit: is this function overriden somewhere? If not, it could be made private